skopt/optimizer/optimizer.py

```diff
@@ -0,0 +1,242 @@
+    def __init__(self, dimensions, base_estimator,
+                 n_random_starts=10, acq_func="EI", acq_optimizer="auto",
+                 random_state=None, n_points=10000, n_restarts_optimizer=5,
+                 xi=0.01, kappa=1.96, n_jobs=1):
+        super(Optimizer, self).__init__()
+        # Arguments that are just stored not checked
+        self.acq_func = acq_func
+        self.rng = check_random_state(random_state)
+        self.kappa = kappa
+        self.models = []
+        self.n_points = n_points
+        self.n_restarts_optimizer = n_restarts_optimizer
+        self.space = Space(dimensions)
+        self.Xi = []
+        self.xi = xi
+        self.yi = []
+
+        self._check_arguments(dimensions, base_estimator,
+                              n_random_starts,
+                              acq_func, acq_optimizer,
+                              n_jobs)
+
+        self.parallel = Parallel(n_jobs=n_jobs)
+
+    def _check_arguments(self, dimensions, base_estimator,
+                         n_random_starts,
+                         acq_func, acq_optimizer,
+                         n_jobs):
+        """Check arguments for sanity."""
+        if not is_regressor(base_estimator):
+            raise ValueError(
+                "%s has to be a regressor." % base_estimator)
+        self.base_estimator = base_estimator
+
+        if n_random_starts < 0:
+            raise ValueError(
+                "Expected `n_random_starts` >= 0, got %d" % n_random_starts)
+        self.n_random_starts = n_random_starts
+
+        self.acq_optimizer = acq_optimizer
+        if acq_optimizer == "auto":
+            warnings.warn("The 'auto' option for the acq_optimizer will be "
+                          "removed in 0.4.")
+            self.acq_optimizer = "lbfgs"
+        elif self.acq_optimizer not in ["lbfgs", "sampling"]:
+            raise ValueError(
+                "Expected acq_optimizer to be 'lbfgs' or 'sampling', "
+                "got %s" % acq_optimizer)
+
+    def ask(self):
+        """Suggest next point."""
+        if len(self.Xi) < self.n_random_starts:
+            return self.space.rvs(random_state=self.rng)[0]
+
+        else:
+            if not self.models:
+                raise ValueError("Random evaluations exhausted and no "
+                                 "model has been fit.")
+            # return point computed from last call to tell()
+            return self._next_x
+
+    def tell(self, x, y, fit_model=True):
+        """Record an observation of the objective function.
+
+        Provide values of the objective function at points suggested by `ask()`
+        or other points. By default a new model will be fit to all
+        observations. The new model is used to suggest the next point at
+        which to evluate the objective. This point can be retrieved by calling
+        `ask()`.
+
+        To add several observations as a batch without fitting a new model
+        set `fit_model` to False.
```

Comment:
The use case I had in mind was one where you know you don't want to fit yet, but don't have a batch (say the user is entering them by hand). Mostly a question of where you do the queuing I guess. You decide, I have no strong feelings.